From: Thorsten Kukuk <kukuk@thkukuk.de>
Date: Mon, 4 Apr 2016 13:51:15 +0000 (-0400)
Subject: Remove des*.c dependencies to glibc
X-Git-Tag: libtirpc-1-0-2-rc3
X-Git-Url: http://git.linux-nfs.org/?p=steved%2Flibtirpc.git;a=commitdiff_plain;h=503ac2e9fa569d95e366766202a7ca840e28b28a

Remove des*.c dependencies to glibc

Our des_impl.c has dependencies to glibc header files
and different arguments then our header file has.
Bring our own code in sync.

Signed-off-by: Thorsten Kukuk <kukuk@thkukuk.de>
Signed-off-by: Steve Dickson <steved@redhat.com>
---

diff --git a/src/des_impl.c b/src/des_impl.c
index c5b7ed6..9dbccaf 100644
--- a/src/des_impl.c
+++ b/src/des_impl.c
@@ -6,7 +6,8 @@
 /* see <http://www.gnu.org/licenses/> to obtain a copy.  */
 #include <string.h>
 #include <stdint.h>
-#include <rpc/rpc_des.h>
+#include <sys/types.h>
+#include <rpc/des.h>
 
 
 static const uint32_t des_SPtrans[8][64] =
diff --git a/tirpc/rpc/des.h b/tirpc/rpc/des.h
index d2881ad..018aa48 100644
--- a/tirpc/rpc/des.h
+++ b/tirpc/rpc/des.h
@@ -82,6 +82,6 @@ struct desparams {
 /*
  * Software DES.
  */
-extern int _des_crypt( char *, int, struct desparams * );
+extern int _des_crypt( char *, unsigned, struct desparams * );
 
 #endif

